### PR TITLE
Add failing deposit with mint tests

### PR DESCRIPTION
**Pull request by mslipper**
Tests two different kinds of failing deposits:

- Garbage data, end
- A call to a contract that reverts.

### CODE REVIEW

File: simulators/optimism/l1ops/deposit_tests.go [hunk 1]
```diff
@@ -270,13 +272,75 @@ func erc20RoundtripTest(t *hivesim.T, env *optimism.TestEnv) {
...
+	// Create garbage data
+	randData := make([]byte, 32)
+	_, err = rand.Read(randData)
+	require.NoError(t, err)
+
+	testData := [][]byte{
+		randData,
+		revertTx.Data(),
+	}
+	mintAmount := big.NewInt(0.5 * params.Ether)
```
Comment (protolambda) [inline simulators/optimism/l1ops/deposit_tests.go:311]: I know it's an untyped constant and produces a non-zero value after multiplication, but it still sketches me out to use floating point math in code that uses integers. We don't want a 0 mint amount.

**Review by protolambda — APPROVED** [review]

File: simulators/optimism/l1ops/deposit_tests.go (hunk 1 above)
Comment (mslipper) [reply simulators/optimism/l1ops/deposit_tests.go:311]: I can make it into a constant that uses int math.

Comment (mslipper) [reply simulators/optimism/l1ops/deposit_tests.go:311]: Will do that as a separate PR though, this is used all over the place.

Comment (mslipper) [reply simulators/optimism/l1ops/deposit_tests.go:311]: Tracked in ENG-2861.

**Pull request merged**